OrbitGl/EventTrack.cpp

```diff
@@ -187,15 +187,53 @@ bool EventTrack::IsEmpty() const {
 }
 
 //-----------------------------------------------------------------------------
-std::string EventTrack::GetSampleTooltip(PickingID id) const {
-  auto SafeGetFormattedFunctionName = [](uint64_t addr) -> std::string {
-    auto it = Capture::GAddressToFunctionName.find(addr);
-    return it == Capture::GAddressToFunctionName.end()
-               ? std::string("<i>") + "???" + "</i>"
-               : it->second;
-  };
+std::string SafeGetFormattedFunctionName(uint64_t addr, int max_line_length) {
+  auto it = Capture::GAddressToFunctionName.find(addr);
+  if (it == Capture::GAddressToFunctionName.end()) {
+    return std::string("<i>") + "???" + "</i>";
+  }
 
-  static const std::string unknown_return_text = "Function call information missing";
+  std::string fn_name = ShortenStringWithEllipsis(it->second, max_line_length);
+  // Simple HTML escaping
+  fn_name = Replace(fn_name, "&", "&amp;");
+  fn_name = Replace(fn_name, "<", "&lt;");
+  fn_name = Replace(fn_name, ">", "&gt;");
+  return fn_name;
+};
+
+//-----------------------------------------------------------------------------
+std::string FormatCallstackForTooltip(std::shared_ptr<CallStack> callstack,
```

Comment:
The same as `SafeGetFormattedFunctionName`: I think this can be declared static.